### ISSUE TITLE
2.21 does not compile on macOS

### ISSUE THREAD

**Issue by prj-** [hypre-space/hypre#394]
```
Building libHYPRE_utilities.dylib ... 
/Users/glci/petsc-hash-pkgs/541917/bin/mpicc -dynamiclib -undefined dynamic_lookup  -o libHYPRE_utilities-2.21.0.dylib F90_HYPRE_error.o F90_HYPRE_general.o HYPRE_handle.o HYPRE_version.o amg_linklist.o binsearch.o exchange_data.o fortran_matrix.o hypre_ap.o hypre_complex.o hypre_error.o hypre_hopscotch_hash.o hypre_merge_sort.o hypre_mpi_comm_f2c.o hypre_prefix_sum.o hypre_printf.o hypre_qsort.o hypre_utilities.o mpistubs.o qsplit.o random.o threading.o timer.o timing.o hypre_cuda_utils.obj hypre_general.obj hypre_handle.obj hypre_memory.obj hypre_omp_device.obj hypre_nvtx.obj -install_name @rpath/libHYPRE_utilities-2.21.0.dylib
clang: error: no such file or directory: 'HYPRE_handle.o'
make[1]: *** [libHYPRE_utilities.dylib] Error 1
make: *** [all] Error 1
```
See bottom of log from https://gitlab.com/petsc/petsc/-/jobs/1320438632/artifacts/browse/arch-ci-osx-cxx-pkgs-opt/lib/petsc/conf/ (I can reproduce this locally as well, it's not specific to the PETSc worker).

**Comment by liruipeng** [comment]
@prj- This problem was due to case insensitive file system on Mac. We have fixed it in the current master. Thanks!

**Comment by prj-** [comment]
Thanks, is this fixed in https://github.com/hypre-space/hypre/pull/385 or some other PR? Any plans to ship 2.21.1 soon, or do we have to stick to 2.20.0 until 2.22.0?

**Comment by liruipeng** [comment]
> Thanks, is this fixed in #385 or some other PR?

#385 

> Any plans to ship 2.21.1 soon, or do we have to stick to 2.20.0 until 2.22.0?

Another release will be made by the end of June

**Comment by prj-** [comment]
Great, we will track that newer version in PETSc then.
